lib/qo/pattern_matchers/pattern_match.rb

```diff
@@ -131,15 +153,59 @@ def self.mixin(destructure: false, as: :match)
         end
       end
 
+      # Whether or not the current pattern match requires a matching branch
+      #
+      # @return [Boolean]
+      def exhaustive?
+        @exhaustive
+      end
+
+      # Whether or not the current pattern match is exhaustive and has a missing
+      # default branch
+      #
+      # @return [Boolean]
+      def exhaustive_no_default?
+        exhaustive? && !@default
+      end
+
+      # Names of all of the available branch names set in `Branching` on
+      # registration of a branch
+      #
+      # @return [Array[String]]
+      def available_branch_names
+        self.class.available_branches.keys
+      end
+
+      # Whether or not all branch types have been provided to the matcher.
+      #
+      # @return [Boolean]
+      def all_branches_provided?
+        available_branch_names == @provided_matchers.uniq
```

Comment:
May fail if you provide branches in order other than order of definition 

```suggestion
        available_branch_names.sort == @provided_matchers.uniq.sort
```

Reply:
I tried to integrate latest changes into my PR, and got this error:

```ruby
     Qo::Exceptions::ExhaustiveMatchMissingBranches:
       Exhaustive match required: pattern does not specify all branches.
         Expected Branches: right, left
         Given Branches:    left, right
```

Reply:
I may jump back to implementing this with sets, silly mistake on my part.